### PR TITLE
update amrvac error printing

### PR DESCRIPTION
**Pull request by Kiradorn**
## PR Summary
Relegates the notification of zero-density to RuntimeWarning to prevent the copious printing of the same error. Expect this fix to be temporary in the event a debug option is preferred.

Addresses the error in issue #2859.

## PR Checklist

- [x] pass `black --check yt/`
- [x] pass `isort . --check --diff`
- [x] pass `flake8 yt/`
- [x] pass `flynt yt/ --fail-on-change --dry-run -e yt/extern`
- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: yt/frontends/amrvac/fields.py [hunk 1]
```diff
@@ -40,10 +41,11 @@ def _velocity(field, data, idir, prefix=None):
 
     mask1 = rho == 0
     if mask1.any():
-        mylog.info(
+        warnings.warn(
             "zeros found in %sdensity, "
             "patching them to compute corresponding velocity field.",
             prefix,
```
Comment (neutrinoceros) [inline yt/frontends/amrvac/fields.py:47]:
```suggestion
            f"zeros found in {prefix}density"
            "patching them to compute corresponding velocity field.",
```

File: yt/frontends/amrvac/fields.py [hunk 2]
```diff
@@ -40,10 +41,10 @@ def _velocity(field, data, idir, prefix=None):
 
     mask1 = rho == 0
     if mask1.any():
-        mylog.info(
-            "zeros found in %sdensity, "
+        warnings.warn(
```
Comment (munkm) [inline yt/frontends/amrvac/fields.py:44]: So, is there a reason this is being switched over to a warning? My impression is that warnings.warn should be used if the user can do something about it, and that logging.warning should be used if the event should be noted but isn't something that can be changed by the user?  

The table [in the logging docs](https://docs.python.org/3/howto/logging.html#when-to-use-logging) about issuing warnings distinguishes these events.

Comment (munkm) [reply yt/frontends/amrvac/fields.py:44]: I see the point in the PR description about the copious printing, but I wonder if there is a solution consistent with the use of logger/warning

Comment (neutrinoceros) [reply yt/frontends/amrvac/fields.py:44]: That's an excellent point I didn't think of. For reference, here's an issue to discuss it more thoroughly https://github.com/yt-project/yt/issues/2865

**Comment by neutrinoceros** [comment]
Closing this since #2859 was resolved more globally in #2871

**Pull request closed without merging**